game-core/src/main/java/games/strategy/engine/data/MutableProperty.java

```diff
@@ -145,6 +146,20 @@ public void resetValue() {
     return ofString(stringSetter, noGetter(), noResetter());
   }
 
+  public static <T> MutableProperty<T> ofMapper(
+      final ThrowingFunction<String, T, Exception> mapper,
+      final ThrowingConsumer<T, Exception> setter,
+      final Supplier<T> getter,
+      final Supplier<T> defaultValue) {
+    return new MutableProperty<>(setter, o -> setter.accept(mapper.apply(o)), getter, () -> {
+      try {
+        setter.accept(defaultValue.get());
+      } catch (final Exception e) {
```

Comment:
We should probably allow an unchecked exception to escape un-wrapped, similar to what we do in `setValue()`, as that's what would happen from a normal resetter.